plugins/database/mongodb/mongodb.go

```diff
@@ -224,11 +222,28 @@ func (m *MongoDB) RevokeUser(ctx context.Context, statements dbplugin.Statements
 		WriteConcern: writeconcern.New(writeconcern.WMajority()),
 	}
 
-	result := client.Database(db).RunCommand(ctx, dropUserCmd, nil)
-	return result.Err()
+	return runCommandWithRetry(ctx, client, db, dropUserCmd)
 }
 
 // RotateRootCredentials is not currently supported on MongoDB
 func (m *MongoDB) RotateRootCredentials(ctx context.Context, statements []string) (map[string]interface{}, error) {
 	return nil, errors.New("root credential rotation is not currently implemented in this database secrets engine")
 }
+
+// runCommandWithRetry runs a command with retry.
+func runCommandWithRetry(ctx context.Context, client *mongo.Client, db string, cmd interface{}) error {
```

Comment:
This behavior seems fine as well, but isn't the same as what the original code did, as it only retried once. Also, if this is the behavior that is desired, note that you can use the `context` parameter to control deadlines (e.g. use `context.WithDeadline(ctx, time.Now().Add(1 * time.Minute)` and pass that to `RunCommand`. The driver will return with `context.DeadlineExceeded` if the timeout expires.

Reply:
The original code was:
```
err = session.DB(mongoCS.DB).Run(createUserCmd, nil)
	switch {
	case err == nil:
	case err == io.EOF, strings.Contains(err.Error(), "EOF"):
		// Call getConnection to reset and retry query if we get an EOF error on first attempt.
		session, err := m.getConnection(ctx)
		if err != nil {
			return "", "", err
		}
		err = session.DB(mongoCS.DB).Run(createUserCmd, nil)
		if err != nil {
			return "", "", err
		}
	default:
		return "", "", err
	}
```

To mimic this, I think you can do:
```
func runCommandWithRetry(ctx context.Context, client *mongo.Client, db string, cmd interface{}) error {
	var err error
	for i := 0; i < 2; i++ {
		err = client.Database(db).RunCommand(ctx, cmd).Err()
		if err == nil {
			return nil
		}

		cmdError, ok := err.(mongo.CommandError)
		if !ok || !cmdError.HasErrorLabel("NetworkError") {
			return err
		}
	}

	// both attempts failed - return last error
	return err
}
```

Reply:
Right, but like I said I think it is desired/makes sense to retry here. I avoided manipulating the context here on purpose since the context is handled by Vault itself and I think it makes sense to don't manipulate that deadlock. 

However, I'm not quite sure if one minute is too long. I guess the Vault/Go HTTP timeout will happen earlier and will probably obfuscate the real error which is returned. @kalafut Any opinions here?

Reply:
Ah, sorry, I misunderstood what you were trying to do. If it is desired to retry all errors, I'll defer the decision about how long the retry period should be to you and @kalafut.

Reply:
I think a minute is reasonable. Isn't the default Vault request timeout 90 seconds? My only suggestion would be that instead of sleep 30, I might start at 5s and back off 30% each time (or something like that).

Reply:
oh, and I just noticed you are sleeping 30 _milliseconds_, not seconds. Seems like quite an aggressive retry if that's not a typo.

Reply:
Makes sense. I changed it to a default backoff time. @kalafut

Reply:
thanks